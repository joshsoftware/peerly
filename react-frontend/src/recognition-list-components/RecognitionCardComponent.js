import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { Row, Col } from "core-components/grid/GridComponent";
import { Card } from "core-components/card/CardComponent";
import { BsCircle } from "core-components/icon/icons";
import ImageComponent from "core-components/image/ImageComponent";
import RecognitionCardHeaderComponent from "recognition-list-components/RecognitionCardHeaderComponent";
import RecognitionTextComponent from "recognition-list-components/RecognitionTextComponent";
import HighFiveComponent from "shared-components/high-five-components/HighFiveComponent";

const CoreValueImage = styled(ImageComponent)`
  border-radius: 20px;
  overflow: hidden;
`;

const HighFive = styled(HighFiveComponent)`
  position: absolute;
  bottom: 50px;
`;

const CardComponent = styled(Card)`
  border-radius: 20px;
  margin: 20px 20px;
  box-shadow: 0px 5px 20px var(--box-shadow-color);
  opacity: 1;
  background: var(--white) 0% 0% no-repeat padding-box;
`;

const RecognitionCardComponent = ({
  givenByName,
  givenByImage,
  givenForName,
  givenForImage,
  givenAt,
  text,
  coreValue,
  coreValueImage,
}) => (
  <CardComponent>
    <Card.Body>
      <Row>
        <Col sm="12" md="7">
          <RecognitionCardHeaderComponent
            givenAt={givenAt}
            givenForName={givenForName}
            coreValue={coreValue}
            givenForImage={givenForImage}
          />
          <CoreValueImage
            src={coreValueImage}
            alt="Core value"
            className="d-sm-block d-xs-block d-md-none mt-2"
          />
          <RecognitionTextComponent
            givenByImage={givenByImage}
            givenByName={givenByName}
            text={text}
          />
        </Col>
        <Col className="d-none d-md-block text-center">
          <div>
            <CoreValueImage
              src={coreValueImage}
              alt="Core value"
              className="mt-2"
            />
          </div>
          <BsCircle size="50px" className="mt-4" color="red" />
        </Col>
      </Row>
      <HighFive />
      <Card.Footer className="bg-white">
        <span className="font-weight-bold text-muted">+</span>
        <span className="font-weight-bold text-dark">1</span>
      </Card.Footer>
    </Card.Body>
  </CardComponent>
);

RecognitionCardComponent.propTypes = {
  givenByName: PropTypes.string.isRequired,
  givenByImage: PropTypes.string,
  givenForName: PropTypes.string.isRequired,
  givenForImage: PropTypes.string,
  givenAt: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  coreValue: PropTypes.string.isRequired,
  coreValueImage: PropTypes.string,
};

RecognitionCardComponent.defaultProps = {
  coreValueImage: "https://i.picsum.photos/id/654/300/200.jpg",
};

export default React.memo(RecognitionCardComponent);
