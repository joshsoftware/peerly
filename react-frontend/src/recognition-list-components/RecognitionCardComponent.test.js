import React from "react";
import { render } from "@testing-library/react";

import RecognitionCardComponent from "recognition-list-components/RecognitionCardComponent";

describe("Recognition Card Component test", () => {
  test("should equal snapshot", () => {
    const { asFragment } = render(
      <RecognitionCardComponent
        givenByName="Roy"
        givenForName="David"
        givenAt="Today 10:30 AM"
        coreValue="positive"
        text="give high five for good work"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
