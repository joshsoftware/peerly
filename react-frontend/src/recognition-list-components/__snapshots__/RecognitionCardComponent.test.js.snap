// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recognition Card Component test should equal snapshot 1`] = `
<DocumentFragment>
  .c2 {
  width: 12vh;
}

.c1 {
  color: var(--black);
  background-color: var(--white);
  font-weight: bold;
  border: none;
}

.c4 {
  border: 1px solid var(--grey);
  border-radius: 11px;
  opacity: 1;
  margin-left: 5%;
  margin-right: 5%;
}

.c5 {
  width: auto;
  margin-left: 10%;
}

.c6 {
  width: 7vh;
  margin-right: 5%;
}

.c3 {
  border-radius: 20px;
  overflow: hidden;
}

.c7 {
  position: absolute;
  bottom: 50px;
}

.c0 {
  border-radius: 20px;
  margin: 20px 20px;
  box-shadow: 0px 5px 20px var(--box-shadow-color);
  opacity: 1;
  background: var(--white) 0% 0% no-repeat padding-box;
}

<div
    class="c0 card"
  >
    <div
      class="card-body"
    >
      <div
        class="row"
      >
        <div
          class="col-md-7 col-sm-12"
        >
          <div>
            <div
              class="d-flex justify-content-end"
            >
              <button
                class="c1 btn btn-primary"
                type="button"
              >
                ...
              </button>
            </div>
            <div
              class="d-flex flex-row"
            >
              <div
                class="c2"
              >
                <img
                  alt="Profile"
                  class="img-fluid rounded-circle"
                  src="https://i.picsum.photos/id/2/200/200.jpg"
                />
              </div>
              <div
                class="text-start ml-2"
              >
                <label
                  class="font-weight-bold  form-label"
                >
                  David
                </label>
                <div>
                  <label
                    class="form-label"
                  >
                     got a high five for
                  </label>
                  <label
                    class="ml-1 font-weight-bold form-label"
                  >
                    positive
                  </label>
                </div>
                <label
                  class="text-muted form-label"
                >
                  Today 10:30 AM
                </label>
              </div>
            </div>
          </div>
          <img
            alt="Core value"
            class="c3 d-sm-block d-xs-block d-md-none mt-2 img-fluid"
            src="https://i.picsum.photos/id/654/300/200.jpg"
          />
          <div>
            <fieldset
              class="c4"
            >
              <legend
                class="c5"
              >
                â€œ
              </legend>
              <label
                class="px-2 pb-2 pt-0 form-label"
              >
                give high five for good work
              </label>
            </fieldset>
            <div
              class="d-flex justify-content-end w-100"
            >
              <label
                class="font-weight-bold text-dark form-label"
              >
                Roy
              </label>
              <img
                alt="Profile"
                class="c6 img-fluid rounded-circle"
                src="https://i.picsum.photos/id/2/200/200.jpg"
              />
            </div>
          </div>
        </div>
        <div
          class="d-none d-md-block text-center col"
        >
          <div>
            <img
              alt="Core value"
              class="c3 mt-2 img-fluid"
              src="https://i.picsum.photos/id/654/300/200.jpg"
            />
          </div>
          <svg
            class="mt-4"
            color="red"
            fill="currentColor"
            height="50px"
            stroke="currentColor"
            stroke-width="0"
            style="color: red;"
            viewBox="0 0 16 16"
            width="50px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8 15A7 7 0 108 1a7 7 0 000 14zm0 1A8 8 0 108 0a8 8 0 000 16z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
      <img
        alt="High five recognition"
        class="c7 img-fluid"
        src="/assets/images//high-five.png"
      />
      <div
        class="bg-white card-footer"
      >
        <span
          class="font-weight-bold text-muted"
        >
          +
        </span>
        <span
          class="font-weight-bold text-dark"
        >
          1
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
